### PR TITLE
Feature/sidebar routing

### PR DESCRIPTION
**Pull request by cyberkade**
## Description

Each user role should have buttons in the sidebar that make sense to their role. These buttons should be organized in a matter where everything is clear and easy to find. This PR adds functionality to Sidebar.js, by which it dynamically renders Links with routes to respective components based on the logged in user's role. 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

## Loom Video

https://www.loom.com/share/ca17cb4ab3004d50b85129e4f9d5765b

### CODE REVIEW

File: src/components/common/Sidebar/SidebarComponents.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  4: <div>"Track Resources" Component goes here</div>,
+  5: <div>"Pending Applications" Component goes here</div>,
+  6: <div>"Manage Users" Component goes here</div>,
+  7: <div>"Support Requests" Component goes here</div>,
+  8: <RenderUpdateProfile />,
+  9: <div>Do we need Account settings?</div>,
+  11: <div>"View Meetings" Component goes here</div>,
+};
+
+export { AdminComponents, MentorComponents, MenteeComponents };
```
Comment (ayv8er) [inline src/components/common/Sidebar/SidebarComponents.js:46]: Thank you for moving the components per role to another file! It was difficult parsing through the Sidebar.js file with all that there.

File: src/components/common/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+  const isUserAdmin = () => {
+    if (role === '2') {
+      return true;
+    }
+    return false;
+  };
+
+  return (
+    <Layout style={{ minHeight: '100vh' }}>
```
Comment (ayv8er) [inline src/components/common/Sidebar/Sidebar.js:65]: I'm loving how you set up conditional component rendering.

**Review by ayv8er — APPROVED** [review]

File: src/components/common/Sidebar/Sidebar.js [hunk 3]
```diff
@@ -184,6 +182,8 @@ const Sidebar = props => {
       <Layout className="site-layout">
         <Content style={{ margin: '2vh 1vw' }}>
           <Content>
+            {/* This is where the component rendering magic happens. Depending on the value of "key" in Menu.Item, the render variable will change,
```
Comment (nhamilton1) [inline src/components/common/Sidebar/Sidebar.js:185]: Love the comments, this helps!

File: src/components/common/Sidebar/SidebarComponents.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import CalendarFeature from '../Calendar';
 import RenderUpdateProfile from '../../pages/UpdateProfile/RenderUpdateProfile';
 
+// These components are being exported to Sidebar.js, from there they are being dynamically rendered based on what role the user is
```
Comment (nhamilton1) [inline src/components/common/Sidebar/SidebarComponents.js:5]: I like what you did here adding SidebarComponents.js

**Review by nhamilton1 — APPROVED** [review]
Overall, great work, I really like the comments you added with your code!

File: src/components/common/Sidebar/Sidebar.js (hunk 3 above)
Comment (rickmansfield) [reply src/components/common/Sidebar/Sidebar.js:185]: perfectly descriptive for those that take the underdog devs project on later. Nicely done.

**Review by rickmansfield — APPROVED** [review]
dynamically rendering Links with routes to respective components based on the logged-in user's role is essential to this project. Thanks for handling this sidebar routing feature.

**Review by Retrofitt — APPROVED** [review]
Fantastic work here , the SidebarComponent.js file cleans up our Sidebar.js really nicely.

File: src/components/common/Sidebar/Sidebar.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+  const handleLogout = checked => {
+    authService.logout();
+    localStorage.removeItem('role_id');
+  };
+
+  const handleMenuClick = menu => {
+    updateRender(menu.key);
+  };
+
+  // This is determining which role is currently in session, implemented further in ternary statements in the return clause
```
Comment (paulstgermain) [inline src/components/common/Sidebar/Sidebar.js:41]: Smartly-written comments, here!

File: src/components/common/Sidebar/Sidebar.js (hunk 2 above)
Comment (paulstgermain) [reply src/components/common/Sidebar/Sidebar.js:63]: Seconding this - this was a really efficient way to implement dynamic rendering!

**Review by paulstgermain — APPROVED** [review]
## Loom Walkthrough Review

I think this is the first time I've ever had a teammate play soothing background music in their PR walkthrough - and I loved it! 😆

The rest of the walkthrough was great as well - you were very thorough, and explained each aspect of your work and what it's doing in a very concise, professional manner. The context I gleaned from this video made it so I was already very familiar with the code by the time I reviewed it, which is exactly what we're looking for!

## Code Review

The code is of equal quality here - you implemented great logic in an easy-to-read, well-formatted manner, and your use of external functions and dynamic rendering to ensure the user only sees what they should see made these components not only reusable but much more efficient as well!

I also want to note that your efforts to clean up these components and add detailed notes and comments so that future devs know exactly what's happening here is greatly appreciated, and something you'll want to do in your upcoming career as well!

**Pull request merged**